railgun/internal/proxy/clientgo_cached_proxy_resolver.go

```diff
@@ -187,6 +219,55 @@ func (p *clientgoCachedProxyResolver) startCacheServer() {
 // Client Go Cached Proxy Resolver - Private Methods - Cache Server Utils
 // -----------------------------------------------------------------------------
 
+// initialize validates connectivity with the Kong proxy and some of the configuration options thereof
+// and populates several local attributes given retrieved configuration data from the proxy root config.
+//
+// Note: this must be run (and must succeed) in order to successfully start the cache server.
+func (p *clientgoCachedProxyResolver) initialize() error {
+	// download the kong root configuration (and validate connectivity to the proxy API)
+	root, err := p.kongRootWithTimeout()
+	if err != nil {
+		return err
+	}
+
+	// pull the proxy configuration out of the root config and validate it
+	proxyConfig, ok := root["configuration"].(map[string]interface{})
+	if !ok {
+		return fmt.Errorf("invalid database configuration, expected a string got %t", proxyConfig["database"])
```

Comment:
Copy pasta of proxyConfig["database"] error message, I guess ?

Reply:
No, but I just realized I messed up this formatter, thanks for helping me spot it!

```suggestion
		return fmt.Errorf("invalid database configuration, expected a string got %T", proxyConfig["database"])
```